pkg/api/handlers/compat/events.go

```diff
@@ -68,24 +77,52 @@ func GetEvents(w http.ResponseWriter, r *http.Request) {
 		}
 	}(r.Context().Done())
 
-	// Headers need to be written out before turning Writer() over to json encoder
-	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusOK)
-	if flusher, ok := w.(http.Flusher); ok {
-		flusher.Flush()
-	}
+	// Write the events from the channel.
+	go func() {
+		first := true
+		var coder *jsoniter.Encoder
+		json := jsoniter.ConfigCompatibleWithStandardLibrary
```

Comment:
Why do we need to redefine this? This should be set for the entire package

Reply:
No idea. I will add a TODO as I consider it to be out of context of this PR. I don't want to change to much in a fix.

Reply:
I was going to say that I was fairly certain it was already set, but it looks like I'm wrong. Which is actually quite concerning, perf-wise - we have a super-efficient JSON encoder/decoder but we're just not using it in the API, which deals heavily with JSON.